src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        Node<T> x;
+        if (index < (size >> 1)) {
+            x = first;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+        } else {
+            x = last;
+            for (int i = size - 1; i > index; i--) {
+                x = x.prev;
+            }
+        }
+        return x;
+    }
+
+    private Node<T> getNode(T value) {
+        Node<T> x;
+        x = first;
+        for (int i = 0; i < size; i++) {
+            if (x.item == value || x.item.equals(value)) {
+                return x;
+            }
+            x = x.next;
+        }
+        return null;
+    }
+
+    private void unlink(Node node) {
+        size--;
+
+        if (node.prev != null) {
+            node.prev.next = node.next;
+        }
+        if (node.next != null) {
+            node.next.prev = node.prev;
+        }
+        if (node.equals(first)) {
```

Comment:
equals не перевизначений для класу node краще його не юзати
```suggestion
        if (node == first) {
```